### PR TITLE
[IMP] generic: create merge_pdf method.

### PR DESCRIPTION
**Pull request by mep-odoo**
Description of the issue/feature this PR addresses:
TASK : https://www.odoo.com/web#id=30639&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720

PAD: https://pad.odoo.com/p/r.ee4121a30256c4b463a2466b411d00fc

### CODE REVIEW

File: odoo/addons/base/ir/ir_actions_report.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def _merge_pdf(documents):
         for document in documents:
             pdfreport = open(document, 'rb')
             streams.append(pdfreport)
-            reader = PdfFileReader(pdfreport, overwriteWarnings=False)
+            reader = PdfFileReader(pdfreport, overwriteWarnings=False, strict=False)  # https://bugs.launchpad.net/pypdf/+bug/242755
```
Comment (hmo-odoo) [inline odoo/addons/base/ir/ir_actions_report.py]: Fedex International deliver provider is giving extra-page in labels pdf. so we need to pass strict params with False value in PdfFileReader

@mep-odoo , can we add argument in _merge_pdf method: like strict=True
and pass strict=False from merge_pdf method of delivery object

File: addons/delivery/models/delivery_carrier.py [hunk 2]
```diff
@@ -188,3 +194,18 @@ def fixed_get_tracking_link(self, pickings):
...
+    def merge_pdf(self, label_name, label_data):
+        pdfdocuments = []
+        for label_binary_data in label_data:
+            pdfreport_fd, pdfreport_path = tempfile.mkstemp(suffix='.pdf', prefix='delivery_label.tmp.')
+
+            # Directly load the document if we already have it
+            with closing(os.fdopen(pdfreport_fd, 'wb')) as pdfreport:
+                pdfreport.write(label_binary_data[1])
+            pdfdocuments.append(pdfreport_path)
+        entire_report_path = ir_actions_report._merge_pdf(pdfdocuments, strict=False)
```
Comment (rim-odoo) [inline addons/delivery/models/delivery_carrier.py]: requires a rebase as this method does not exist anymore

File: addons/delivery/models/delivery_carrier.py [hunk 3]
```diff
@@ -187,3 +189,16 @@ def fixed_get_tracking_link(self, picking):
 
     def fixed_cancel_shipment(self, pickings):
         raise NotImplementedError()
+
+    def merge_pdf(self, label_name, label_data):
```
Comment (rim-odoo) [inline addons/delivery/models/delivery_carrier.py]: Can you make this method a bit generic and move it to `odoo/tools/pdf.py` (to be created)?
Actually, it has no link with `delivery.carrier` objects, it is *used by* `delivery.carrier`s

Comment (rim-odoo) [inline addons/delivery/models/delivery_carrier.py]: Also document input variables: we should not have to guess that data is a list of pairs

File: addons/delivery/models/delivery_carrier.py [hunk 4]
```diff
@@ -187,3 +189,16 @@ def fixed_get_tracking_link(self, picking):
...
+        pdfwriter = PdfFileWriter()
+        for label_binary_data in label_data:
+            reader = PdfFileReader(StringIO(label_binary_data[1]), strict = False)
+            for page in range(0, reader.getNumPages()):
+                pdfwriter.addPage(reader.getPage(page))
+        output = StringIO()
+        pdfwriter.write(output)
+        output_value = output.getvalue()
+        output.close()
+        labels = [('Label%s.pdf'%(label_name), output_value)]
```
Comment (rim-odoo) [inline addons/delivery/models/delivery_carrier.py]: Why returning a list? It will ALWAYS be a list with 1 tuple

**Pull request merged**